src/components/Button.tsx

```diff
@@ -1,62 +1,92 @@
 import { AriaButtonProps } from "@react-types/button";
+import { PressEvent } from "@react-types/shared";
 import { ReactNode, RefObject, useMemo, useRef } from "react";
 import { useButton, useFocusRing, useHover } from "react-aria";
-import { Icon, IconProps } from "src";
+import { Link } from "react-router-dom";
+import { Icon, IconProps, navLink } from "src";
 import { Tooltip } from "src/components/Tooltip";
 import { Css } from "src/Css";
 import { BeamButtonProps, BeamFocusableProps } from "src/interfaces";
+import { isAbsoluteUrl, noop } from "src/utils";
 
-export interface ButtonProps extends BeamButtonProps, BeamFocusableProps {
+export interface ButtonProps extends Omit<BeamButtonProps, "onClick">, BeamFocusableProps {
   label: string;
   variant?: ButtonVariant;
   size?: ButtonSize;
   icon?: IconProps["icon"];
   endAdornment?: ReactNode;
   /** HTML attributes to apply to the button element when it is being used to trigger a menu. */
   menuTriggerProps?: AriaButtonProps;
-  buttonRef?: RefObject<HTMLButtonElement>;
+  buttonRef?: RefObject<HTMLElement>;
+  /** If function, then it is the handler that is called when the press is released over the target. Otherwise if string, it is the URL path for the link */
+  onClick?: ((e: PressEvent) => void) | string;
```

Comment:
Moved the `onClick` definition here to support `string` as well as a callback function. I haven't put it into `BeamButtonProps` interface as that is also used by IconButton. However, if we think we'd want to support URLs/paths on IconButton, then once the implementation of this is flushed out via discussions on this PR, then I'm happy to add support of links to IconButtons as well.

Reply:
Added to IconButton as well.